README.md

```diff
@@ -125,3 +123,16 @@ completes (or fails).
 When the `PipelineRun` has completed, the `taskRuns` field will contain
 references to all `TaskRuns` which were executed and their next and
 previous `TaskRuns`.
+
+### Resources
+
+`Resources` in a pipelines are the set of objects that are going to be used 
+as inputs and outputs of a `TaskRun`. 
+
+* `Resources` is created directly in a pipeline configuration and bound 
+to `TaskRun` as an input and/or output source. 
+* The (optional) `passedConstraint` key on an `input source` defines a set of previous task names.
+* When the `passedConstraint` key is specified on an input source, only the version of 
+the resource that passed through the defined list of tasks is used.
+* The `passedConstraint` allows for `Tasks` to fan in and fan out, and ordering can be expressed explicitly 
+using this key since a task needing a resource from a another task would have to run after.
```

Comment:
Any interest in updating the README Pipeline overview image to the one you presented today @shashwathi @pivotal-nader-ziada ? And adding the resources v. pipelineparams image?

(I'm happy to add these if you'd prefer)

Reply:
The same diagrams are in the PR branch. Should be updated. Feel free to add them in any other location you prefer